src/OrbitGl/App.cpp

```diff
@@ -2641,7 +2641,7 @@ orbit_base::Future<std::vector<ErrorMessageOr<void>>> OrbitApp::ReloadModules(
     // loaded by the process.
     if (!process->IsModuleLoadedByProcess(module->file_path())) {
       RemoveFrameTrack(func);
-    } else if (!module->is_loaded()) {
+    } else if (!module->AreAtLeastFallbackSymbolsLoaded()) {  // TODO: Is this correct?
```

Comment:
I'm not very clear about this part.

But the `frame_track_function_hashes_map[module_to_reload->file_path()]` is used later in `EnableFrameTracksFromHashes`.

Now `AddFallbackSymbol` is not called to update the `hash_to_function_map_`.
So for all the fallback symbols we added here, we will get the following error message in log
```
ORBIT_ERROR("Could not find function hash %#x in module \"%s\"", function_hash,
                  module->file_path());
```
later as they are not found in `hash_to_function_map_`

Reply:
Ok, so:

> Now AddFallbackSymbol is not called to update the hash_to_function_map_.

I'm not sure what you mean here. Yes, `AddFallbackSymbol` is not called anywhere yet because it's not yet integrated (next PR or so), but when it is you can imagine it as the same as `AddSymbols`.

> So for all the fallback symbols we added here, we will get the following error message in log

Unless I'm missing something, this shouldn't be the case. On reload, if the loaded symbols were the "fallback symbols" before the reload and they still are, re-hooking based on hashing should work normally. In the uncommon case that the symbols before the reload were the debug symbols and after they are the "fallback symbols", or viceversa, you will probably get the error (unless there is a match between a function from the debug symbols and a function from the dynamic linking symbols of the "fallback symbols"), but at that point that error is correct.